### PR TITLE
Updated C#/.NET compatibility tables. (DOCS-11023)

### PR DESCRIPTION
**Pull request by rob-mongodb**
Staged at https://docs-mongodbcom-staging.corp.mongodb.com/ecosystem/robertjustice/DOCS-11023/drivers/csharp.html.

### CODE REVIEW

**Comment by rob-mongodb** [comment]
@schmalliso , good catch! Staging is updated and ready for another look.

File: source/drivers/csharp.txt [hunk 1]
```diff
@@ -127,6 +134,12 @@ MongoDB Compatibility
      - .NET 4.5
      - .NET Core
 
+   * - Version 2.4
```
Comment (schmalliso) [inline source/drivers/csharp.txt]: This isn't appearing on staging, currently -- not sure why. Assuming it does build correctly, LGTM!

**Comment by schmalliso** [comment]
One comment - LGTM assuming it all builds correctly!

**Comment by rob-mongodb** [comment]
@jyemin , this small update is ready for a technical review.

**Comment by jyemin** [comment]
After discussing with @rstam, we decided it would be good to explicitly mention all the versions of .NET Core, as we do with .NET Framework. So the table would look like:

```
Driver Version ...        .NET Core 1.0 | .NET Core 1.1 | .NET Core 2.0
2.4                                x    |         x     |        x 
2.3                                x    |         x     |        x 
```

**Comment by rob-mongodb** [comment]
@jyemin , staging is updated and ready for another look.

File: source/drivers/csharp.txt [hunk 2]
```diff
@@ -125,37 +132,57 @@ MongoDB Compatibility
      - .NET 3.5
      - .NET 4.0
      - .NET 4.5
-     - .NET Core
+     - .NET Core 1.1
```
Comment (jyemin) [inline source/drivers/csharp.txt]: Re-order so that 1.0 is before 1.1

**Comment by rob-mongodb** [comment]
@kay-kim , this is ready for merge into 3.6 and master.

**Pull request merged**